### ISSUE TITLE
Fastboot Extension Issue

### ISSUE THREAD

**Issue by deanmraz** [gcollazo/ember-cli-showdown#55]
# Issue: 
- When running with fastboot and using custom extension, I get this error below: 

```
Error: Error in myextensionname extension-> sub-extension 0: property "type" must be a string, but undefined given
```
# Notes 
- works fine when you run it without fastboot. 

# Example Project
- https://github.com/deanmraz/ember-showdown 
- just run yarn install and ember serve 

I will be looking for solution but just wanted to open this issue to see if anyone has already solved it!

**Comment by deanmraz** [comment]
@gcollazo let me know if you have any thoughts.

**Comment by jasonmit** [comment]
I'll take a look as soon as I can.  Thank you for the example, will speed up the process.

**Comment by jasonmit** [comment]
@deanmraz just an update, looks like showdown is taking a wrong path:
https://github.com/showdownjs/showdown/blob/b5e46fdb8dd72ddc74c532e80410536c0a38523c/src/showdown.js#L195

Showdown is wrapping an array within an array because isArray(ext) === false when it should be true.

![image](https://user-images.githubusercontent.com/3108309/29730586-8d02c62e-8995-11e7-9d94-df234d39c2e5.png)

This is likely an issue with the use of Fastboot's use of Node's VM API that the Ember app is run within.  Speculation... need to dig deeper.

**Comment by jasonmit** [comment]
https://github.com/nodejs/node/issues/855#issuecomment-74583712 I believe is the root of the problem.  The work around for now is do not return an array from within your extension function.

**Comment by jasonmit** [comment]
https://github.com/showdownjs/showdown/pull/425 is the fix, will need to wait for it to be merged/versioned.

**Comment by jasonmit** [comment]
Reopening since I have a patch on our end.  Minor breaking change so will version as 4.x.  The breaking change is no longer rely on the showdown global and require it to be imported.

**Comment by jasonmit** [comment]
Released as 4.0.0

**Comment by deanmraz** [comment]
@jasonmit confirmed it works in my project, thanks man great work!
